JSONLD examples with working scripts.

wolfiex opened the pull request:
### LD repository: 
- individual_ld.jsonfiles as identified by their id. 
- Frame.jsonld file to outline format and to be used as a template for framing. 
- Schema.jsonld 
- Version.jsonld (for all the files) 
- Graph.jsonld to group all entries, and define their origin.

durack1 commented on JSONLD/auxillary/mip_era/cmip5.json:
```diff
@@ -0,0 +1,8 @@
+    "@id": "mip-cmor-tables:auxillary/mip_era/cmip5",
+    "mip:name": "CMIP5",
+    "mip:url": "https://wcrp-cmip.org/CMIP5",
+    "mip:start": "2999",
```
Did we want these to be real numbers?

wolfiex replied:
And potentially the correct dates too.

durack1 commented on JSONLD/auxillary/mip_era/cmip6.json:
```diff
@@ -0,0 +1,8 @@
+    "@id": "mip-cmor-tables:auxillary/mip_era/cmip6",
+    "mip:name": "CMIP6",
+    "mip:url": "https://wcrp-cmip.org/CMIP6",
+    "mip:start": "2999",
```
Ditto here, planning started 2013, first data July 2018, publication continuing today

wolfiex replied:
is there a planning close?

durack1 replied:
Planning and reality are not the same thing, but yes, I would assume that after 2025 there will be zero additional publications to the project, we might even manage to get this done before the end of 2024, but we'll see - requires the ESGF nodes and SOLR index management involvement to do this

durack1 commented on JSONLD/auxillary/mip_era/cmip6plus.json:
```diff
@@ -0,0 +1,8 @@
+    "@id": "mip-cmor-tables:auxillary/mip_era/cmip6plus",
+    "mip:name": "CMIP6Plus",
+    "mip:url": "https://wcrp-cmip.org/CMIP6Plus",
+    "mip:start": "2999",
```
Planning 2023, first publication 2024-01-02

durack1 commented on JSONLD/consortia/context.json:
```diff
@@ -0,0 +1,8 @@
+    "consortia": {
+        "@context": {
+            "@vocab": "consortia",
+            "@base": "https://github.com/PCMDI/mip-cmor-tables/tree/json-ld/JSONLD/consortia"
```
is the json-ld/JSONLD path duplication needed?

wolfiex replied:
one is the branch name. This file has been replaced.

durack1 commented on JSONLD/consortia/csiro-arccss.json:
```diff
@@ -0,0 +1,18 @@
+                "@id": "mip-cmor-tables/institutions/csiro"
+            },
+            {
+                "@id": "ARCCSS (Australian Research Council Centre of Excellence for Climate System Science)"
```
Is this @id and @id duplicated key with differing contents (path to csiro) and descriptive string correct?

wolfiex replied:
One is a path to a file we have, the other is an unknown (see missing.txt)

durack1 commented on JSONLD/consortia/dlr-bira.json:
```diff
@@ -0,0 +1,18 @@
+                "@id": "Deutsches Zentrum fur Luft- und Raumfahrt"
+            },
+            {
+                "@id": "Royal Belgian Institute for Space Aeronomy"
```
This follows a different format all descriptive strings rather than a path and then a descriptive string - compared to JSONLD/consortia/csiro-arccss.json

wolfiex replied:
This follows the text string with no acronym format. Also known as missing data.

wolfiex replied:
Moved to incomplete.

durack1 reviewed (commented):
@wolfiex great to see this starting to coalesce! With > 5000 file changes, not sure exactly how best to provide a "review" other than a couple of the tidbits that I have started to look at. Is there anything more specifically that I can do to provide feedback, and/or could be partition the review tasks across @taylor13, @matthew-mizielinski and myself - or even open this up to more to peruse at the WIP telco tomorrow?

durack1 commented on JSONLD/organisations/consortia/ucsd-sio.json:
```diff
@@ -8,11 +8,34 @@
-                "@id": "University of California, San Diego"
+                "@type": "consortia:member",
+                "consortia:institution": {
+                    "@id": "mip-cmor-tables:organisations/institutions/university"
```
This looks a little wrong

durack1 commented on JSONLD/organisations/consortia/ucsd-sio.json:
```diff
@@ -8,11 +8,34 @@
+                    {
+                        "from": 2016,
+                        "to": 2022,
+                        "phase": "CMIP6"
```
This is an obs4MIPs contributor alone

durack1 commented on JSONLD/organisations/consortia/solaris-heppa.json:
```diff
@@ -0,0 +1,374 @@
+            "consortia:membership_type": "current",
+            "consortia:dates": [
+                {
+                    "from": 2025,
```
Keeping track of this will be hard, and to simplify the task, I would omit. Briony and Beth might be good to consult on this TBH?

durack1 commented on JSONLD/organisations/consortia/solaris-heppa.json:
```diff
@@ -0,0 +1,374 @@
+                {
+                    "from": 2025,
+                    "to": "-",
+                    "phase": "CMIP7"
```
NCAR is a contributor to AMIP1, AMIP2, CMIP1, CMIP2, CMIP2+, CMIP3, [CMIP4 which has never been formally identified], CMIP5, CMIP6, and plans to contribute to CMIP7, probably CMIP8, ... just to consider how we'd use this field

durack1 commented on JSONLD/organisations/consortia/espri-ipsl.json:
```diff
@@ -8,11 +8,34 @@
+                "consortia:membership_type": "prior",
+                "consortia:dates": [
+                    {
+                        "from": 2016,
```
IPSL first data were published 2018-07-02 (ESGF index entry), but they were involved well before that on ES-DOC etc, so not sure how we define this start and end time

durack1 commented on JSONLD/organisations/consortia/espri-ipsl.json:
```diff
@@ -8,11 +8,34 @@
-                "@id": "Ensemble de Services Pour la Recherche"
+                "@type": "consortia:member",
+                "consortia:institution": {
+                    "@id": "mip-cmor-tables:organisations/institutions/ensemble"
```
Not sure this makes sense, at least to me

durack1 reviewed (commented):
Just peeking this latest iteration and dropping a couple of comments to ponder

wolfiex commented:
@durack1 I am happy for this to finally be over! 

I have created an output [MIP_organisations.json](https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json) which follows our file structure and combines a simplified consortium and institution list. 

Is there anything else needed before merging?

durack1 reviewed (commented):
Just trying to review the entries in JSONLD/scripts/output/MIP_organisations.json I am having trouble finding this in a branch/PR to review/comment on

durack1 commented:
Couple of nits, it looks like the wrong institution is listed here

https://github.com/PCMDI/mip-cmor-tables/blob/e22640c2399f367840eb970b70658b3b0f248644/JSONLD/scripts/output/MIP_organisations.json#L43C1-L48C7

"Beijing Biocytogen (China)" -> "Beijing Climate Center" https://ror.org/03kb2jw75

durack1 commented:
Did we want to ascii (0-127 char) wash these, they will not likely play nice with HTML

https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json#L85-L90

"Centre National d'Études Spatiales" -> "Centre National dEtudes Spatiales"

durack1 commented:
https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json#L133-L138

"Freie Universität Berlin" -> "Freie Universitat Berlin",

durack1 commented:
https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json#L265-L270

Also NCAS `"name": "Uks",`

durack1 commented:
https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json#L307-L312

"name": "Lawrence Livermore National Laboratory", -> "name": "Program for Climate Model Diagnosis and Intercomparison",

There is already an `LLNL` defined, with same ROR

wolfiex commented on JSONLD/archive/showcase /developement/prepare_institutions.ipynb:
It will eventually be deleted.

wolfiex commented:
> Couple of nits, it looks like the wrong institution is listed here
> 
> https://github.com/PCMDI/mip-cmor-tables/blob/e22640c2399f367840eb970b70658b3b0f248644/JSONLD/scripts/output/MIP_organisations.json#L43C1-L48C7
> 
> "Beijing Biocytogen (China)" -> "Beijing Climate Center" https://ror.org/03kb2jw75

I would like to fix this using a acton one the two have been merged. We should delete the file, and flag an issue.

wolfiex commented:
> https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/scripts/output/MIP_organisations.json#L331-L336
> 
> "name": "Remote Sensing Systems (United States)", -> "name": "Remote Sensing Systems"

Hoping that we could flag issues on individual files.

durack1 commented:
> Hoping that we could flag issues on individual files not as part of a pull request for something else.

Sure, maybe we can catch these afterward - with me tagged to do it - as a way of figuring out how specific little issues can be done by someone, not you.. I see 4 discrete issues, each with a specific institution, with each that could be captured as a single issue for me to process

wolfiex commented:
- Clarified the source JSON files for user interaction. 
- migrated JSONLD-ification to processing and checks scripts. 
- Added more metadata to JSONLD
- removed it for simple view located in https://github.com/PCMDI/mip-cmor-tables/tree/jsonld/JSONLD/scripts/output folder

durack1 commented on .github/libs/checks/__pycache__/schema.cpython-310.pyc:
couple of these to nix

durack1 commented on .github/libs/checks/__pycache__/institution.cpython-310.pyc:
another

durack1 commented on .github/libs/checks/__pycache__/__init__.cpython-310.pyc:
another

durack1 commented on .github/libs/checks/.DS_Store:
another

durack1 commented on .github/libs/__pycache__/action_functions.cpython-310.pyc:
another

durack1 commented on JSONLD/archive/showcase /combine_ld.py:
is the space in the path `archive/showcase /combine..` expected?

wolfiex replied:
Yes, but left because that section will be deleted at some point

durack1 commented on JSONLD/archive/showcase /developement/prepare_institutions.ipynb:
`archive/showcase /developement..` both space and typo in "development" -> development

durack1 replied:
and also 
BCC - not Beijing Biocytogen (China), rather "Beijing Climate Centre" https://ror.org/03kb2jw75
ESSO - right acronym? Not INCOIS
NCAS - institution name 'Uks' is wrong, should be "National Centre for Atmospheric Science" https://ror.org/01wwwe276
SOLARIS-HEPPA - is a consortium, not institution

wolfiex replied:
Deleted the inccorect ones, Will re-add using the issues template once merged. 

https://github.com/PCMDI/mip-cmor-tables/issues/56 

Can you  clarify the ESSO issue there?

Solaris heppa only appears within the consortium section. I did not see it in the institution files.

durack1 replied:
> Can you clarify the ESSO issue there?

This is in the notebook

```python
  'ESSO': {'identifiers': {'institution_name': 'Indian National Centre for Ocean Information Services',
    'aliases': [],
    'acronyms': ['INCOIS'],
```

CMIP-IPO replied:
These are all issues that will be addressed in the edit procedure to follow after the pull request

durack1 commented on JSONLD/archive/showcase /sample_generated/latlon_data.json.bak:
nix?

durack1 reviewed (commented):
Caught a couple of quick things to fix, there might be others

durack1 commented:
@wolfiex just found another nit, I think I couldn't see this before as GitHub only loads the first 3000 files, and this must've been 3001 or more.

The question, in the file name `solaris-heppa.json` vs the file contents `..consortia/solaris_heppa` - so the underscore vs the hyphen. No idea if this is an issue or not, but seemed to be an inconsistency

https://github.com/PCMDI/mip-cmor-tables/blob/1343761411e9ae21b309c4789c57a70a223c43f8/JSONLD/organisations/consortia/solaris-heppa.json#L1-L8

durack1 commented:
Looks like the `JSONLD/miptables/variables/` subdirectories both contain duplicate *.bak files

matthew-mizielinski reviewed (approved):
I believe the changes here are all confined to the JSONLD and .github actions directories.

As things stand this doesn't impact on the CMOR 3.x formatted mip tables nor does it alter the controlled vocabulary entries at the root of this repository.

From my testing with the CMIP-LD repository code this change supports the construction of the CVs for the CMIP6Plus repository, and I've been able to test the resulting CMIP6Plus_CVs.json file.

One item I did note in scanning over the files under JSONLD is the spelling of developement (change to development if possible).

wolfiex commented:
Latest commits added to re-generate the main repository files - see commits above. These are identical in the diff, just with different 'version metadata' . 

The running of these scripts still needs to be incorporated into the action, but happy to merge in the meantime. 

@durack1 @matthew-mizielinski

durack1 commented on JSONLD/auxillary/mip-era/cmip5.json:
```diff
@@ -0,0 +1,8 @@
+  "@type": "mip-era",
+  "end": 2999,
+  "name": "CMIP5",
+  "start": 2999,
```
Planning began in 2006, last project publication was 20200617

durack1 commented on JSONLD/auxillary/realm/land.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "@id": "mip-cmor-tables:auxillary/realm/land",
+  "@type": "realm",
+  "description": "Land Surface",
```
This also includes subsurface, so soil moisture etc

durack1 reviewed (approved):
Adding a couple of nits for consideration

wolfiex commented:
Resolved changes suggested by @durack1

durack1 commented:
I'm happy to press that big green button if you want me to! This is great, almost there..
